hw_9.py

```diff
@@ -0,0 +1,105 @@
+"""Homework_9"""
+
+# # # Строки с заданным символом # # #
+# Напишите программу, которая бы работала следующим образом -
+# находила символ "#" и если этот символ найден -
+# удаляла предыдущий символ из строки.
+# Ваша задача обработать строки с "#" символом.
+
+
+def string_modifier(string: str) -> str:
+    """Функция замены символа перед #"""
+    if "".join(string.split("#")) == "":
+        return ""
+    parts = string.split("#")
+    result = ""
+    for part in parts:
+        if len(result) == 0:
+            result += part
+        else:
+            result = result[:-1] + part
+    return result
+
+
+assert string_modifier('a#bc#d') == 'bd'
+assert string_modifier('abc#d##c') == 'ac'
+assert string_modifier('abc##d######') == ''
+assert string_modifier('#######') == ''
+assert string_modifier('') == ''
+assert string_modifier('abcde') == 'abcde'
+
+
+# # # Свечи # # #
+# Когда свеча догорает, остается остаток.
+# Остатки можно объединить, чтобы создать новую свечу,
+# которая при догорании,
+# в свою очередь, оставит еще один остаток.
+# У вас есть количество свечей.
+# Какое общее количество свечей вы можете сжечь, если предположить,
+# что вы создадите новые свечи, как только у вас останется достаточно остатков?
+# Пример
+# Если у Вас 5(candles_number) свечей,
+# и из 2х(make_new) остатков вы можете сделать 1 новую свечу,
+# то ответе будет: 9.
+# По шагам, чтобы сжечь 9 свечей:
+# сожгите 5 свечей, получите 5 остатков;
+# создайте еще 2 свечи, используя 4 остатка (остался 1);
+# сожгите 2 свечи, в итоге останется 3 остатка;
+# создайте еще одну свечу, используя 2 остатка (остался 1);
+# сожгите созданную свечу, что даст еще один остаток (всего 2 остатка);
+# создать свечу из оставшихся остатков;
+# зажгите последнюю свечу.
+# Таким образом, можно сжечь 5+2+1+1=9 свечей, что и является ответом.
+new_candles_init = 0
+make_new = 0
+
+
+def burned_candles_counter(new_candles, make_new_candle):
```

Comment:
if you decided yo use typehints, pls use it everywhere

Reply:
Sorry, it's not clear what you mean: are you about variable names?

Reply:
`def burned_candles_counter(new_candles: <type>, make_new_candle: <type>) -> <type>:`

Reply:
fixed